modules/gdprEnforcement.js

```diff
@@ -183,7 +189,7 @@ export function deviceAccessHook(fn, gvlid, moduleName, result) {
           gvlid = getGvlid(moduleName) || gvlid;
         }
         const curModule = moduleName || curBidder;
-        let isAllowed = validateRules(purpose1Rule, consentData, curModule, gvlid);
+        let isAllowed = validateRules(purpose1Rule, consentData, curModule, gvlid, moduleType);
```

Comment:
is it possible to check for purpose1 consent to the publisher or purpose 1 in general (instead of on a specific vendor) here, as per https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#tcdata

Reply:
I was going to suggest that @patmmccann, but with a caveat. TCF CMPs may, but are not required to, provide transparency and consent for TCF Purposes _for the Publisher_ and then appropriately reflect those signals in the tcstring and API response data. So checking those signals makessense when the site has configured their CMP to disclose the publisher's processing.

Reply:
Not sure I follow you guys. We have a `validatRules` function, where I've added additional logic to not check the vendor consent for vendorless module: https://github.com/prebid/Prebid.js/blob/afe88d862f031631058f343622e487af6bb4d900/modules/gdprEnforcement.js#L151
Do you, mean, that this exception should be done only if `purposeId === 1`?

Reply:
I apologize for the confusion. 

Perhaps we can see best in an additional unit test. Sharedid module (after the vendor code is deleted) should be allowed to use storage manager if there is purpose 1 consent. and should not be allowed if there is not.

Reply:
gotcha will do

Reply:
Thanks!